scripts/two-node-local-net.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+
+# run a two node local net
+
+set -e
+
+PROJECT_ROOT=$(git rev-parse --show-toplevel)
+# shellcheck disable=SC1090
+source "$(dirname "$0")"/common.sh
+
+cd "$PROJECT_ROOT"
+
+last_modified_rust_file=$(
+  find . -path ./target -prune -o -type f -name '*.rs' -printf '%T@ %p\n' |
+  sort -nr |
+  head -1 |
+  cut -d' ' -f2-
+)
+
+polkadot="target/release/polkadot"
+
+# ensure the polkadot binary exists and is up to date
+if [ ! -x "$polkadot" ] || [ "$polkadot" -ot "$last_modified_rust_file" ]; then
+  cargo build --release
+fi
+
+# setup variables
+node_offset=0
+declare -a node_pids
+declare -a node_pipes
+
+# create a sed expression which injects the node name and stream type into each line
+function make_sed_expr() {
+  name="$1"
+  type="$2"
+
+  printf "s/^/%8s %s: /" "$name" "$type"
+}
+
+# turn a string into a flag
+function flagify() {
+  printf -- '--%s' "$(tr '[:upper:]' '[:lower:]' <<< "$1")"
+}
+
+# start a node and label its output
+#
+# This function takes a single argument, the node name.
+# The name must be one of those which can be passed to the polkadot binary, in un-flagged form.
+function run_node() {
+  name="$1"
+  # create a named pipe so we can get the node's PID while also sedding its output
+  local stdout
+  local stderr
+  stdout=$(mktemp --dry-run --tmpdir)
+  stderr=$(mktemp --dry-run --tmpdir)
+  mkfifo "$stdout"
+  mkfifo "$stderr"
+  node_pipes+=("$stdout")
+  node_pipes+=("$stderr")
+
+  # compute ports from offset
+  local port=$((30333+node_offset))
+  local rpc_port=$((9933+node_offset))
+  local ws_port=$((9944+node_offset))
+  node_offset=$((node_offset+1))
```

Comment:
I guess this will work only for up to 11 nodes, probably fine, but maybe worth documenting?

Reply:
You're right that it's limited to 11 nodes, but there's another, stricter limit: this only works properly if the node name is one of the default nodes: alice, bob, charlie, dave, eve, ferdie, one, two. We can't get to the 11 limit even by manually editing the script.